addons/stock/models/product.py

```diff
@@ -559,7 +559,6 @@ class ProductTemplate(models.Model):
 
     responsible_id = fields.Many2one(
         'res.users', string='Responsible', default=lambda self: self.env.uid, company_dependent=True, check_company=True,
-        domain="['|', ('company_id', '=', False), ('company_ids', 'in', allowed_company_ids[0])]",
```

Comment:
just test your situation 1 and it's working at my end

this should work, the in in the domain is working à l'envers

Reply:
The `in` works, but it would only allow to select a user in the current company, which means you wouldn't have all the valid choices available.

Reply:
yeah but in this specific case responsible_id is a property field so the value will be saved for the current company so it makes sense to filter the others out doesn't it?

Reply:
But i'll verify that in [False] or in [] doesn't block the search.

Reply:
I missed the company_dependent :man_facepalming:. Forget about this case, it does indeed work fine :D.